staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -141,6 +141,7 @@ func createAuditEventAndAttachToContext(req *http.Request, policy audit.PolicyRu
 	}
 
 	req = req.WithContext(request.WithAuditEvent(ctx, ev))
+	req = req.WithContext(audit.WithAuditPolicyRuleEvaluator(req.Context(), policy))
```

Comment:
WithContext is allocating a new http.Request object.

Instead of doing that twice we should do:
```
ctx = request.WithAuditEvent(ctx, ev)
ctx = audit.WidthAuditPolicyRuleEvaluator(ctx, policy)
req = req.WithContext(ctx)
``

Reply:
done